### DISCUSSION TITLE
Is it able to change the Round Name

### DISCUSSION THREAD

**Discussion by MeiYien16** [Drarig29/brackets-viewer.js#93] [Q&A] [answered]
Hi, Is it able to change the Round Name from Round 1, Round 2 to Quarter Finals, Semi Finals. Somemore, for the case BYE, is it able to change the BYE copywriting to 'No opponent'?

**Comment by Drarig29** [comment, marked as answer]
> Is it able to change the Round Name from Round 1, Round 2 to Quarter Finals, Semi Finals.

No it wasn't possible to do so yet, because I'm relying on `i18next`, which doesn't support what I wanted.

I just added a new config property for this, named `customRoundName`.
You can use it like this (🚧 **edit:** `info.group` was renamed to `info.groupType`):

https://github.com/Drarig29/brackets-viewer.js/blob/ed31fc4fc43336d3543411f802a8b1d9d592d467/demo/with-api.html#L46-L52

For Quarter Finals, then the condition will be `info.fractionOfFinal === 1 / 4`.

And the result is:

<img width="401" alt="image" src="https://user-images.githubusercontent.com/9317502/230772697-b64dd543-b4d0-4d6d-b6e3-0ebded03bbbc.png">

---

> For the case BYE, is it able to change the BYE copywriting to 'No opponent'?

Yes, you can change it by overriding a single key of [`./src/i18n/en/translation.json`](https://github.com/Drarig29/brackets-viewer.js/blob/master/src/i18n/en/translation.json#L89).

Example:

```js
window.bracketsViewer.addLocale('en', {
    "common": {
        "bye": "No opponent",
    }
});
```

**Reply by MeiYien16** [reply]
I just tried the `customRoundName`, found that for winner bracket Grand Final, it show GF Round 1, seems like grand final not included in round count. Is there any way to edit grand final name so? Thanks!

**Reply by Drarig29** [reply]
Oh thanks, you found a bug in the naming + I didn't think about final rounds at this time.

I renamed `info.group` into `info.groupType` because I didn't like the naming. And I added a `info.finalType`. You can find the type definition here:

https://github.com/Drarig29/brackets-viewer.js/blob/4c9c685c897d4e98dc44119cfbd91d6f90fe856d/src/types.d.ts#L110-L130

And a demo of how to use `customRoundName` with final rounds:

https://github.com/Drarig29/brackets-viewer.js/blob/4c9c685c897d4e98dc44119cfbd91d6f90fe856d/demo/with-api.html#L46-L60

This was released in [`v1.5.1`](https://github.com/Drarig29/brackets-viewer.js/releases/tag/v1.5.1).

**Reply by MeiYien16** [reply]
Thanks ya, I tested the grand finalType == 'grand-final', it works well!
The finalType == 'consolation-final' seems like affected by latest changes, it now default show 'Round 1' & 'Final Round' for roundNumber 1 & 2. 
Could you add fractionOfFinal in groupType == 'final-group' so? It would be helpful on customization. 
Appreciate it!

**Reply by Drarig29** [reply]
It doesn't make sense to have `fractionOfFinal` in the final.

In a consolation final, you only have 1 round.

And in a grand final, you have a maximum of 2 rounds, but **with the same number of opponents** in both rounds. So the first round is not a "semi grand final round".

**Reply by MeiYien16** [reply]
I see, got it! Thanks a lot!

**Reply by Drarig29** [reply]
> The finalType == 'consolation-final' seems like affected by latest changes, it now default show 'Round 1' & 'Final Round' for roundNumber 1 & 2.

I don't see any problem here. Consolation final should not have anything to do with roundNumber 1 & 2.

![image](https://user-images.githubusercontent.com/9317502/231681056-05f4fa1f-23d7-4d24-bca4-f9189379a06e.png)

**Reply by Drarig29** [reply]
And consider giving a star to the projects if you like them! (manager & viewer :grin:)

**Comment by MeiYien16** [comment]
Thanks for your answer, how about disable the match hover effect ya?

**Reply by Drarig29** [reply]
Why?

**Reply by MeiYien16** [reply]
I just found out it could be disabled by set `highlightParticipantOnHover` to false.

**Reply by Drarig29** [reply]
Oh okay, I didn't remember this option TBH 😁

**Comment by MeiYien16** [comment]
Hi, now facing another issue, hope you could help🙏 For double elimination, if the participant number is not 2^n, Lower bound if the match have no opponent, the other participant for the match wont proceed to next game like upper bound

**Reply by Drarig29** [reply]
If you still have the issue, please open a new issue.
